recipe/meta.yaml

```
@@ -1,49 +1,70 @@
-{% set version = "2.4.2" %}
+{% set name = "nlopt" %}
+{% set version = "2.7.1" %}
 
 package:
-  name: nlopt
+  name: {{ name }}
   version: {{ version }}
 
 source:
-  fn: nlopt-{{ version }}.tar.gz
-  url: http://ab-initio.mit.edu/nlopt/nlopt-{{ version }}.tar.gz
-  md5: d0b8f139a4acf29b76dbae69ade8ac54
+  url: https://github.com/stevengj/{{ name }}/archive/v{{ version }}.tar.gz
+  sha256: db88232fa5cef0ff6e39943fc63ab6074208831dc0031cf1545f6ecd31ae2a1a
+  patches:
+    - pep3149.patch
 
 build:
-  number: 2
-  features:
-    - vc9  # [win and py27]
-    - vc14  # [win and py>=35]
+  number: 0
+  run_exports:
+    - {{ pin_subpackage('nlopt', max_pin='x.x') }}
+  missing_dso_whitelist:
+    - "$RPATH/ld64.so.1"  # [s390x]
 
 requirements:
   build:
     - {{ compiler('c') }}
     - {{ compiler('cxx') }}
-    - autoconf
-    - automake
-    - libtool
-    - cmake         # [win]
+    - cmake
+    - make      # [unix]
     - swig
-    - python
-    - numpy 1.8.*   # [not (win and (py35 or py36))]
-    - numpy 1.9.*   # [win and py35]
-    - numpy 1.11.*  # [win and py36]
-
-  run:
-    - python
-    - numpy >=1.8   # [not (win and (py35 or py36))]
-    - numpy >=1.9   # [win and py35]
-    - numpy >=1.11  # [win and py36]
+    - m2-patch  # [win]
+    - patch     # [not win]
+  # nlopt is used by r-nloptr so a Python interface isn't needed on win64
+  host:           # [not win]
+    - python      # [not win]
+    - numpy       # [not win]
+  run:            # [not win]
+    - python      # [not win]
+    - {{ pin_compatible('numpy') }}  # [not win]
 
 test:
-  imports:
-    - nlopt
+  imports:        # [not win]
+    - nlopt       # [not win]
+  commands:
+    - test -f $PREFIX/include/nlopt.h                         # [unix]
+    - test -f $PREFIX/include/nlopt.hpp                       # [unix]
+    - test -f $PREFIX/lib/pkgconfig/nlopt.pc                  # [unix]
+    - test -f $PREFIX/lib/libnlopt.so.0                       # [linux]
+    - test -f $PREFIX/lib/libnlopt.so.0.11.1                  # [linux]
+    - test -f $PREFIX/lib/libnlopt.so                         # [linux]
+    - test -f $PREFIX/lib/libnlopt.0.11.1.dylib               # [osx]
+    - test -f $PREFIX/lib/libnlopt.0.dylib                    # [osx]
+    - test -f $PREFIX/lib/libnlopt.dylib                      # [osx]
+    - if not exist %PREFIX%/Library/include/nlopt.h exit 1    # [win]
+    - if not exist %PREFIX%/Library/include/nlopt.hpp exit 1  # [win]
+    - if not exist %PREFIX%/Library/lib/nlopt.lib exit 1      # [win]
+    - if not exist %PREFIX%/Library/bin/nlopt.dll exit 1      # [win]
 
 about:
-  home: http://ab-initio.mit.edu/nlopt
-  license: LGPL 2.1
+  home: https://nlopt.readthedocs.io
+  doc_url: https://nlopt.readthedocs.io
+  dev_url: https://github.com/stevengj/nlopt/
+  license: LGPL-2.1-or-later AND MIT
```

```suggestion
  license: LGPL-2.1-or-later AND MIT AND BSD-3-Clause
```
https://github.com/stevengj/nlopt/tree/v2.7.1/src/algs/slsqp is BSD-3-clause

---

Thanks! I've extended `license_file` by adding all explicit licenses in the `src/algs` folder:
```
  license_file:
    - COPYING
    - src/algs/cobyla/COPYRIGHT
    - src/algs/direct/COPYING
    - src/algs/esch/COPYRIGHT
    - src/algs/luksan/COPYRIGHT
    - src/algs/newuoa/COPYRIGHT
    - src/algs/slsqp/COPYRIGHT
    - src/algs/stogo/COPYRIGHT
```